File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        # Using a manual implementation for finding the first occurrence
        n, m = len(haystack), len(needle)
        if m == 0:
            return 0  # Edge case: empty needle
        
        for i in range(n - m + 1):
            if haystack[i:i + m] == needle:
                return i
        return -1

File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.cpp
#include <string> 

using namespace std; 

 

class Solution { 

public: 

    int strStr(string haystack, string needle) { 

        // Find the position of the first occurrence of needle in haystack 

        size_t pos = haystack.find(needle); 

         

        // If found, return the position; otherwise, return -1 

        return (pos != string::npos) ? pos : -1; 

    } 

}; 